.github/workflows/lint.yml

```
@@ -0,0 +1,26 @@
+name: Lint python projects
+
+on:
+  pull_request: {}
+  workflow_dispatch: {}
+
+jobs:
+  lint:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.x
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.x'
+    - name: Install black and ruff
+      run: pip install black ruff
```

```suggestion
    - name: Install ruff
      run: pip install ruff
```
Sorry, should have mentioned earlier ruff takes care of both formatting and linting. See https://docs.astral.sh/ruff/formatter/

Consider using our pytemplate which has pre-commit also configured for both: https://github.com/CQCL/pytemplate/blob/main/.pre-commit-config.yaml